test/framework/src/main/java/org/elasticsearch/node/NodeTests.java

```diff
@@ -165,6 +174,106 @@ public void testNodeAttributes() throws IOException {
+        final Path defaultPathData = createTempDir().toAbsolutePath();
+        final Settings settings = Settings.builder()
+                .put("path.home", home)
+                .put("path.data.0", zero)
+                .put("path.data.1", one)
+                .put("default.path.data", defaultPathData)
+                .put("http.enabled", false)
+                .put("transport.type", "mock-socket-network")
+                .build();
+        Files.createDirectories(defaultPathData.resolve("nodes/0"));
+        final boolean indexExists = randomBoolean();
+        if (indexExists) {
+            final int numberOfIndices = randomIntBetween(1, 3);
+            for (final String index :
+                    IntStream.range(0, numberOfIndices).mapToObj(i -> UUIDs.randomBase64UUID()).collect(Collectors.toList())) {
+                Files.createDirectories(defaultPathData.resolve("nodes/0/indices").resolve(index));
+            }
+        }
+        Supplier<MockNode> constructor = () -> new MockNode(settings, Collections.singletonList(MockTcpTransportPlugin.class));
+        if (indexExists) {
+            final IllegalStateException e = expectThrows(IllegalStateException.class, constructor::get);
+            final String message = String.format(
+                    Locale.ROOT,
+                    "detected index data in default.path.data [%s] where there should not be any",
+                    defaultPathData.resolve("nodes/0/indices"));
+            assertThat(e, hasToString(containsString(message)));
+        } else {
+            try (Node ignored = constructor.get()) {
+                // node construction should be okay
+            }
+        }
+    }
+
+    public void testDefaultPathDataSet() throws IOException {
+        final Path zero = createTempDir().toAbsolutePath();
+        final Path one = createTempDir().toAbsolutePath();
+        final Path defaultPathData = createTempDir().toAbsolutePath();
+        final Settings settings = Settings.builder()
+                .put("path.home", "/home")
+                .put("path.data.0", zero)
+                .put("path.data.1", one)
+                .put("default.path.data", defaultPathData)
+                .build();
+        try (NodeEnvironment nodeEnv = new NodeEnvironment(settings, new Environment(settings))) {
+            final boolean indexExists = randomBoolean();
+            final List<String> indices;
+            if (indexExists) {
+                final int numberOfIndices = randomIntBetween(1, 3);
+                indices = IntStream.range(0, numberOfIndices).mapToObj(i -> UUIDs.randomBase64UUID()).collect(Collectors.toList());
+                for (final String index : indices) {
+                    Files.createDirectories(nodeEnv.defaultNodePath().indicesPath.resolve(index));
+                }
+            } else {
+                indices = Collections.emptyList();
+            }
+            final Logger mock = mock(Logger.class);
```

Comment:
Might as well `spy(logger)` instead of creating a new mocked instance? At least then you'll see the logging message(s) in the test output